pyteal/ast/abi/reference_type.py

```diff
@@ -0,0 +1,103 @@
+from abc import abstractmethod
+from typing import cast
+from pyteal.ast.abi.uint import Uint, UintTypeSpec
+
+
+class ReferenceTypeSpec(UintTypeSpec):
```

Comment:
I'm not really sure what purpose this class and `ReferenceType` serve. They complicate the inheritance hierarchy so I think there should be a good justification for their presence. Do you mind elaborating on why they're necessary?

Reply:
I had originally done it to keep the bit size setting to 8 in a single spot but since both you and @ahangsu both mentioned it, I've taken it out.

Reply:
Sounds good. Resolving.